examples/advanced/main.tf

```diff
@@ -0,0 +1,102 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.49.0"
+    }
+    azapi = {
+      source = "Azure/azapi"
+    }
+  }
+}
+provider "azurerm" {
+  features {}
+
+  skip_provider_registration = true
+}
+
+provider "azapi" {
+}
+
+resource "azurerm_resource_group" "deploy" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_virtual_network" "deploy" {
+  name                = "deploy-vnet"
+  location            = azurerm_resource_group.deploy.location
+  resource_group_name = azurerm_resource_group.deploy.name
+  address_space       = ["10.0.0.0/16"]
+}
+
+resource "azurerm_subnet" "deploy" {
+  name                 = "deploy-subnet"
+  resource_group_name  = azurerm_resource_group.deploy.name
+  virtual_network_name = azurerm_virtual_network.deploy.name
+  address_prefixes     = ["10.0.1.0/24"]
+}
+
+# The following network security rules are required for entra domain services to work.
```

Comment:
Shouldn we make sure, that these rules exist and note that in the readme, if they are required?